typegen/src/typegen/error.rs

```diff
@@ -68,3 +74,80 @@ pub enum TypeSubstitutionErrorKind {
     #[error("Cannot find matching param on 'from' type.")]
     NoMatchingFromType,
 }
+
+/// Error attempting to do type substitution.
+#[derive(Debug, thiserror::Error, Default, PartialEq, Eq)]
+pub struct SettingsValidationError {
+    /// Cannot add derive for type that is not present in the registry.
+    pub derives_for_unknown_types: Vec<(syn::Path, HashSet<syn::Path>)>,
+    /// Cannot add attribute for type that is not present in the registry.
+    pub attributes_for_unknown_types: Vec<(syn::Path, HashSet<syn::Attribute>)>,
+    /// Invalid path to replace: the type getting substituted is not present in the registry.
+    pub substitutes_for_unknown_types: Vec<(syn::Path, syn::Path)>,
+}
+
+impl std::fmt::Display for SettingsValidationError {
```

Comment:
nit; It would be interesting to see if we could provide alternative types. Since we cannot find type `a::b::Ty`, we could inspect the type registry for `Ty` and return all valid paths. 

Which for an xcm MultiLocation it would return both v2 and v3 entries, which might make things easier during development. Tho, this could probably be a follow-up

Reply:
I have a feeling that Subxt won't be able to use the function that produces this error, so it may well be that it's not that useful for us actually, and instead we'll lean on those other `validation.rs` functions, one of which does provide a list of similar paths

Reply:
Yeah, the `similar_type_paths_in_registry` function can do that, but I have not put it into the `SettingsValidationError`, to not bloat it too much. We should definitely do this though from the subxt macro side.